### PR TITLE
support for recording audio

### PR DESCRIPTION
**Pull request by shmuelhizmi**
support for recording room audio

### CODE REVIEW

**Comment by shmuelhizmi** [comment]
closing due to a bug (you are not self-recording)

**Comment by shmuelhizmi** [comment]
I'm also having troubles since I merge staging to my branch due to #579 
@benawad

**Comment by shmuelhizmi** [comment]
I fixed the bug, a few notes:
1 - currently there is no icon for the recording button.
2 - users volume is not applied to the recording, I can fix that by setting the volume of the users inside the track and not in the audio element like it is now, I just wasn't sure if there is a specific reason we set the volume in the element and not the track.
3 - currently I just make the browser download the audio file the second the recording is done, do we prefer to ask the user first?
4 - do we want the recording file name to say something specific? maybe `[the_room_name].mp3` ?

File: kofta/src/app/components/BottomVoiceControl.tsx [hunk 1]
```diff
@@ -177,6 +179,34 @@ export const BottomVoiceControl: React.FC<BottomVoiceControlProps> = ({
...
+
+		if (recorder.isRecording) {
+      buttons.push(
+        <button
+          className={buttonStyle}
+          key="stop-recording"
+          onClick={() => {
+            recorder.toggle();
+          }}
+          title="Stop Recording"
```
Comment (gitaumoses4) [inline kofta/src/app/components/BottomVoiceControl.tsx]: The title could be translated as well.

File: kofta/src/app/components/BottomVoiceControl.tsx [hunk 2]
```diff
@@ -177,6 +179,34 @@ export const BottomVoiceControl: React.FC<BottomVoiceControlProps> = ({
...
+      );
+    } else {
+      buttons.push(
+        <button
+          className={buttonStyle}
+          key="start-recording"
+          onClick={() => {
+            recorder.toggle();
+          }}
+          title="Start Recording"
```
Comment (gitaumoses4) [inline kofta/src/app/components/BottomVoiceControl.tsx]: Same here.

**Comment by benawad** [comment]
see  #578 for notes, this may be hard to do atm

**Comment by cdbattags** [comment]
I think we can start [here](https://github.com/ethand91/mediasoup3-record-demo) as an example?

https://github.com/ethand91/mediasoup3-record-demo

**Pull request closed without merging**